dashboard/app/models/user.rb

```diff
@@ -816,6 +816,14 @@ def oauth_only?
     end
   end
 
+  def has_google_oauth_matching_primary_email?
+    if migrated?
+      authentication_options.any? {|ao| ao.google_oauth? && ao.email == email}
```

Comment:
I don't believe I put this requirement in this Jira Task, but could we make this more restrictive and only allow a single Authentication Option (`google_oauth`)?  That is a requirement of some of the related Tasks in this Story.

Going further down this line of reasoning, could we consolidate some of the login in the ZenDesk sos controller into a  method on `User` called `employee?` (or something similar like `internal_cdo?`).  That method would require:

- User email domain = code.org
- only authentication option is Google auth
- User email = authentication option email

Reply:
:+1: An `employee?` check sounds like a great idea.  I think it might need to be even more strict than you describe though, if we want to potentially exclude facilitators and/or contributors we've granted `code.org` addresses to.

Reply:
If we can't easily distinguish between employees and contributors when someone authenticates on Code Studio using Google OAuth, then a method that identifies the broader set of internal code.org users would suffice, for now (`internal_cdo?`).

Reply:
My concern with making google SSO the only option is that I don't know of a good way to remove a password if one is associated with an account (I couldn't create a query that finished in any reasonable amount of time to actually find out if this is an issue).

There are a handful of code.org employees but I do feel better about telling them to migrate, though I don't know why they aren't. I thought we migrated over most users...

I'll refactor the functions today!

Reply:
I pulled a report of all employee logins and whether they have Google OAuth enabled.  I'll need to work with all of the people on that list to update their accounts before we merge this functionality.  I'll sync up with you on that this afternoon.